main.tf

```diff
@@ -85,22 +84,29 @@ data "aws_route53_zone" "default" {
 ## A Route 53 record contains authoritative DNS information for a specified DNS name. DNS records are most commonly used to map a name to an IP Address..
 ##----------------------------------------------------------------------------------
 resource "aws_route53_record" "default" {
-  count = var.enable && var.enable_dns_validation ? 1 : 0
+  for_each = {
+    for record in aws_acm_certificate.cert[0].domain_validation_options[*] : record.domain_name => {
+      name   = record.resource_record_name
+      record = record.resource_record_value
+      type   = record.resource_record_type
+    }
+  }
 
-  zone_id         = join("", data.aws_route53_zone.default[*].zone_id)
-  ttl             = var.ttl
   allow_overwrite = var.allow_overwrite
-  name            = join("", aws_acm_certificate.cert[*].domain_validation_options[*].resource_record_name)
-  type            = join("", aws_acm_certificate.cert[*].domain_validation_options[*].resource_record_type)
-  records         = [join("", aws_acm_certificate.cert[*].domain_validation_options[*].resource_record_value)]
+  name            = each.value.name
+  records         = [each.value.record]
+  ttl             = var.ttl
+  type            = each.value.type
+  zone_id         = join(",", data.aws_route53_zone.default[*].zone_id)
 }
 
 ##----------------------------------------------------------------------------------
 ## Most commonly, this resource is used together with aws_route53_record and aws_acm_certificate to request a DNS validated certificate, deploy the required validation records and wait for validation to complete.
 ##----------------------------------------------------------------------------------
 resource "aws_acm_certificate_validation" "default" {
-  count = var.enable && var.enable_dns_validation ? 1 : 0
-
-  certificate_arn         = join("", aws_acm_certificate.cert[*].arn)
-  validation_record_fqdns = aws_route53_record.default[*].fqdn
+  count                   = var.enable && var.enable_dns_validation ? 1 : 0
+  certificate_arn         = join(",", aws_acm_certificate.cert[*].arn)
```

Comment:
Why we are using join function here? Also [*] can be replaced with [0] 'coz count for certificate resource is set to 1.

Reply:
because list of string is requird 
![image](https://github.com/clouddrove/terraform-aws-acm/assets/152850912/2f636d16-cb77-4b17-adc7-faaa18844eb9)

Reply:
than we can skip using join function.

Reply:
yes, we can skip join function. Thank you..!!